aeron-cluster/src/main/java/io/aeron/cluster/ConsensusModule.java

```diff
@@ -3971,6 +3971,27 @@ public TimerServiceSupplier timerServiceSupplier()
             return timerServiceSupplier;
         }
 
+        /**
+         * Registers a ConsensusModuleExtension to extend beahviour of
+         * consensus module instead of using ClusteredServices
+         *
+         * @param extensionSupplier supplier for schema adapter
+         * @return this for a fluent API.
+         */
+        public Context consensusModuleExtension(final Supplier<ConsensusModuleExtension> extensionSupplier)
+        {
+            consensusModuleExtension = extensionSupplier.get();
```

Comment:
New instance should be setup in context conclude. Supplier can be set as a property or loaded dynamically.